### PR TITLE
gh-124 Add Dialog Component To Examples

### PR DESCRIPTION
**Pull request by oleksii-yurchenko**
- create ConfirmDialog
- put ConfirmDialog on ExamplePage

### CODE REVIEW

File: apps/web-portal-client/lib/components/dialog/Dialog.jsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+
+import {IconButton} from './IconButton'
+import {ExitIcon} from './ExitIcon'
+
+export function Dialog({open, onClose, children}) {
+  if (!open) {
+    return <></>
```
Comment (yzhbankov) [inline apps/web-portal-client/lib/components/dialog/Dialog.jsx]: return null

File: apps/web-portal-client/lib/components/dialog/Dialog.jsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  }
+  return (
+    <div className="fixed inset-0 z-50 overflow-auto bg-smoke-light flex">
+      <div className="relative p-8 bg-white w-full max-w-md m-auto flex-col flex rounded-lg">
+        <div>{children}</div>
+        <span className="absolute top-0 right-0 p-4">
+          <IconButton onClick={() => onClose()}>
```
Comment (yzhbankov) [inline apps/web-portal-client/lib/components/dialog/Dialog.jsx]: why we need this "x" button?
you can add a closing dialog by clicking outside the dialog and make this optional (on/off by props passing)

**Comment by yzhbankov** [comment]
looks good

File: apps/web-portal-client/lib/components/dialog/ExitIcon.jsx [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import React from 'react'
+
+export function ExitIcon() {
```
Comment (nesterone) [inline apps/web-portal-client/lib/components/dialog/ExitIcon.jsx]: Could we reuse the already existing <Image> component for that ?

File: apps/web-portal-client/lib/components/dialog/IconButton.jsx [hunk 4]
```diff
@@ -0,0 +1,21 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+
+export function IconButton({onClick, children, className}) {
```
Comment (nesterone) [inline apps/web-portal-client/lib/components/dialog/IconButton.jsx]: > Could we reuse the already existing  component for that ?

same here

**Review by nesterone — CHANGES_REQUESTED** [review]

File: apps/web-portal-client/lib/components/Button.jsx [hunk 5]
```diff
@@ -4,22 +4,24 @@ import PropTypes from 'prop-types'
...
+    <button type="button" className={`btn-general ${addClasses}`} onClick={handleClick}>
       {children || label}
     </button>
   )
 }
 
 Button.propTypes = {
+  addClasses: PropTypes.string,
   label: PropTypes.string,
   handleClick: PropTypes.func,
```
Comment (nesterone) [inline apps/web-portal-client/lib/components/Button.jsx]: handleClick -> onClick

File: apps/web-portal-client/lib/components/dialog/ConfirmDialog.jsx [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+            }}
+          />
+        </div>
+      </div>
+    </Dialog>
+  )
+}
+
+ConfirmDialog.propTypes = {
+  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node]),
```
Comment (nesterone) [inline apps/web-portal-client/lib/components/dialog/ConfirmDialog.jsx]: it makes sense to extract that expression `PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node])` to separate file like `lib/utils/childrenPropTypes.js` and use it in all components where need to define `children` Props

File: apps/web-portal-client/lib/components/dialog/Dialog.jsx [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+            <Image shape={ImageShapes.cross} />
+          </Button>
+        </span>
+      </div>
+    </div>
+  )
+}
+
+Dialog.propTypes = {
+  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node]),
```
Comment (nesterone) [inline apps/web-portal-client/lib/components/dialog/Dialog.jsx]: see comment regarding extraction of the prop type definition

**Review by nesterone — CHANGES_REQUESTED** [review]

File: apps/web-portal-client/lib/components/Button.jsx [hunk 8]
```diff
@@ -1,26 +1,29 @@
 import React from 'react'
 import PropTypes from 'prop-types'
+import {childrenPropTypes} from '../utils/childrenPropTypes'
 
 import './button.css'
 
 export function Button(props) {
-  const {label, handleClick, children} = props
+  const {label, onClick, children, addClasses} = props
```
Comment (yzhbankov) [inline apps/web-portal-client/lib/components/Button.jsx]: "addClasses" sounds like a verb but it is not. plz use something like "cssClass" or "className"

File: apps/web-portal-client/lib/components/dialog/ConfirmDialog.jsx [hunk 9]
```diff
@@ -0,0 +1,52 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import {childrenPropTypes} from '../../utils/childrenPropTypes'
+
+import {Dialog} from './Dialog'
+import {Button} from '../Button'
+
+export function ConfirmDialog({children, title, open, onClose, onConfirm}) {
+  if (!open) return <></>
```
Comment (yzhbankov) [inline apps/web-portal-client/lib/components/dialog/ConfirmDialog.jsx]: return just null

File: apps/web-portal-client/lib/components/dialog/ConfirmDialog.jsx [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+
+  return (
+    <Dialog open={open} onClose={onClose}>
+      <h2 className="text-xl">{title}</h2>
+      <div className="py-5">{children}</div>
+      <div className="flex justify-end">
+        <div className="p-1 w-1/5 flex">
+          <Button
+            label="No"
+            cssClass="bg-smooth-green hover:bg-smooth-green-light flex-auto"
```
Comment (nesterone) [inline apps/web-portal-client/lib/components/dialog/ConfirmDialog.jsx:19]: It's not the best practice to pass `custom` css to such a component like `Button`

We have to standardize button types like, for example:  `<Button type="primary" label="Yes"  />`. We can start with two types of` primary`/`default`, then component would choose to which color to provide for primary and default


Another good practice for dialogs is to highlight the acton button (in confirm dialog it's `Yes`) with primary type and `Cancel` as default type

Comment (nesterone) [reply apps/web-portal-client/lib/components/dialog/ConfirmDialog.jsx:19]: extracting to separate issue

**Review by nesterone — APPROVED** [review]

**Pull request merged**